src/diskio/mod.rs

```diff
@@ -418,9 +422,10 @@ pub(crate) fn write_file_incremental<P: AsRef<Path>, F: Fn(usize)>(
             let len = contents.len();
             // Length 0 vector is used for clean EOF signalling.
             if len == 0 {
+                drop(contents);
```

Comment:
Probably should add a 
`trace_scope!("EOF_chunk", "name": path_display, "len": len);` at the top of this scope.